### PR TITLE
ARROW-16019: [C++] Minimize chances of Minio connect errors

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

File: cpp/src/arrow/filesystem/s3fs_test.cc [hunk 1]
```diff
@@ -100,15 +100,18 @@ class ShortRetryStrategy : public S3RetryStrategy {
...
-           (attempted_retries * kRetryInterval < kMaxRetryDuration);
+    return IsRetryable(error) && (attempted_retries * kRetryInterval < kMaxRetryDuration);
   }
 
   int64_t CalculateDelayBeforeNextRetry(const AWSErrorDetail& error,
                                         int64_t attempted_retries) override {
     return kRetryInterval;
   }
 
+  bool IsRetryable(const AWSErrorDetail& error) {
```
Comment (lidavidm) [inline cpp/src/arrow/filesystem/s3fs_test.cc:111]: nit: `const`?

**Review by lidavidm — APPROVED** [review]

**Comment by pitrou** [comment]
Hmm, the test fails with a timeout on macOS.

**Comment by lidavidm** [comment]
The failures here look unrelated, though, do we already have a JIRA for arrow-threading-utility-test timing out? (It would be nice if we knew which test it timed out on…)

**Review by lidavidm — APPROVED** [review]

**Comment by lidavidm** [comment]
ARROW-15710 is the issue for arrow-threading-utility-test

**Pull request closed without merging**